sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobAsyncClient.java

```diff
@@ -0,0 +1,435 @@
+     * @throws NullPointerException thrown if the {@code tagOrDigest} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<OciManifest> downloadManifest(String tagOrDigest) {
+        return this.downloadManifestWithResponse(tagOrDigest).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Download the manifest associated with the given tag or digest.
+     * We currently only support downloading OCI manifests.
+     *
+     * @see <a href="https://github.com/opencontainers/image-spec/blob/main/manifest.md">Oci Manifest Specification</a>
+     *
+     * @param tagOrDigest The tag or digest of the manifest.
+     * @return The response for the manifest associated with the given tag or digest.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code tagOrDigest} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<OciManifest>> downloadManifestWithResponse(String tagOrDigest) {
+        return withContext(context -> this.downloadManifestWithResponse(tagOrDigest, context));
+    }
+
+    Mono<Response<OciManifest>> downloadManifestWithResponse(String tagOrDigest, Context context) {
+        if (tagOrDigest == null) {
+            return monoError(logger, new NullPointerException("'tagOrDigest' can't be null."));
+        }
+
+        return this.registriesImpl.getManifestWithResponseAsync(repositoryName, tagOrDigest, UtilsImpl.OCI_MANIFEST_MEDIA_TYPE, context)
+            .flatMap(response -> {
+                String digest = response.getHeaders().getValue(UtilsImpl.DOCKER_DIGEST_HEADER_NAME);
+                ManifestWrapper wrapper = response.getValue();
+
+                // The service wants us to validate the digest here since a lot of customers forget to do it before consuming
+                // the contents returned by the service.
+                if (Objects.equals(digest, tagOrDigest) || Objects.equals(response.getValue().getTag(), tagOrDigest)) {
+                    OciManifest ociManifest = new OciManifest()
+                        .setAnnotations(wrapper.getAnnotations())
+                        .setConfig(wrapper.getConfig())
+                        .setLayers(wrapper.getLayers())
+                        .setSchemaVersion(wrapper.getSchemaVersion());
+
+                    Response<OciManifest> res = new ResponseBase<Void, OciManifest>(
+                        response.getRequest(),
+                        response.getStatusCode(),
+                        response.getHeaders(),
+                        ociManifest,
+                        null);
+
+                    return Mono.just(res);
+                } else {
+                    return monoError(logger, new ServiceResponseException("The digest in the response does not match the expected digest."));
```

Comment:
Why would the service return a digest that doesn't match the requested digest?

Reply:
Docker explicitly states that one should check for the digest before consuming the bytes for both security, image corruption etc. 

Service specifically wanted us to add this check since a lot of customers do not validate.